inference-engine/src/mkldnn_plugin/nodes/batch_to_space.cpp

```diff
@@ -3,219 +3,247 @@
-            // TODO: remove Const layers
-            for (int i = 0; i < batchToSpaceLayer->insData.size(); i++) {
-                auto inData = batchToSpaceLayer->insData[i].lock();
-                if (inData == nullptr)
-                    IE_THROW() << "'" << batchToSpaceLayer->name << "' layer has nullable input data";
-                config.inConfs[i].desc = TensorDesc(precision,
-                        inData->getTensorDesc().getDims(),
-                        inData->getTensorDesc().getLayout());
-            }
+                IE_THROW() << "BatchToSpace layer with name '" << batchToSpaceLayer->name << "' has unsupported precision: " << precision.name();
+
+            blockShapeIn = batchToSpaceLayer->_block_shape;
+            cropsBeginIn = batchToSpaceLayer->_crops_begin;
+
+            auto createConfig = [&](Layout layout) {
+                LayerConfig config;
+                // TODO: remove Const layers
+                for (int i = 0; i < batchToSpaceLayer->insData.size(); i++) {
+                    auto inData = batchToSpaceLayer->insData[i].lock();
+                    if (!inData)
+                        IE_THROW() << "BatchToSpace layer with name '" << batchToSpaceLayer->name << "' has nullable input data";
+                    DataConfig inConfig;
+                    if (i == 0)
+                        inConfig.desc = TensorDesc(precision, inData->getTensorDesc().getDims(), layout);
+                    else
+                        inConfig.desc = TensorDesc(inData->getPrecision(), inData->getTensorDesc().getDims(), inData->getTensorDesc().getLayout());
+                    config.inConfs.push_back(inConfig);
+                }
 
-            DataConfig outConfig;
-            outConfig.desc = TensorDesc(precision,
-                    out_dims,
-                    layer->outData[0]->getTensorDesc().getLayout());
-            config.outConfs.push_back(outConfig);
-            config.dynBatchSupport = false;
-            confs.push_back(config);
+                DataConfig outConfig;
+                outConfig.desc = TensorDesc(precision, outDims, layout);
+                config.outConfs.push_back(outConfig);
+
+                config.dynBatchSupport = false;
+                confs.push_back(config);
+            };
+
+            createConfig(inDims.size() == 4 ? NHWC : NDHWC);
+            createConfig(TensorDesc::getLayoutByDims(inDims));
+
+            std::vector<std::pair<ConfLayout, ConfLayout>>  blockConfs { };
+            if (inDims[1] % 8 == 0)  blockConfs.push_back({ConfLayout::BLK8, ConfLayout::BLK8});
+            if (inDims[1] % 16 == 0) blockConfs.push_back({ConfLayout::BLK16, ConfLayout::BLK16});
+            for (auto conf : blockConfs) {
+                addConfig(layer, {DataConfigurator(conf.first, precision),
+                                  DataConfigurator(ConfLayout::ANY, batchToSpaceLayer->insData[1].lock()->getPrecision()),
```

Comment:
I don't think ANY layout is suitable here. Should be: TensorDesc::getLayoutByDims(dims)